atsc/src/header.rs

```diff
@@ -14,19 +14,72 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+use std::panic;
+
 use bincode::{Decode, Encode};
+use log::{debug, trace};
+use version_compare::{compare, Cmp};
 
-#[derive(Encode, Decode, Debug, Clone)]
+#[derive(Encode, Debug, Clone)]
 pub struct CompressorHeader {
     initial_segment: [u8; 4],
+    pub version: String,
     // We should go unsigned
     frame_count: i16,
 }
 
+impl Decode for CompressorHeader {
+    fn decode<__D: ::bincode::de::Decoder>(
+        decoder: &mut __D,
+    ) -> Result<Self, ::bincode::error::DecodeError> {
+        let header = Self {
+            initial_segment: Decode::decode(decoder)?,
+            version: Decode::decode(decoder)?,
+            frame_count: Decode::decode(decoder)?,
+        };
+        let current_version = env!("CARGO_PKG_VERSION").to_string();
+        trace!("Versions: c:{} h:{}", current_version, header.version);
+        match compare(current_version.clone(), header.version.clone()) {
```

Comment:
Nit: Referring to `version-compare` [source code](https://gitlab.com/timvisee/version-compare/-/blob/master/src/compare.rs?ref_type=heads#L29) it looks like it is possible to avoid cloning by passing references to versions:
```suggestion
        match compare(&current_version, &header.version) {
```